opslevel/resource_opslevel_scorecard.go

```diff
@@ -0,0 +1,166 @@
+package opslevel
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opslevel/opslevel-go/v2023"
+)
+
+func resourceScorecard() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a scorecard",
+		Create:      wrap(resourceScorecardCreate),
+		Read:        wrap(resourceScorecardRead),
+		Update:      wrap(resourceScorecardUpdate),
+		Delete:      wrap(resourceScorecardDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The scorecard's name.",
+				ForceNew:    false,
+				Required:    true,
+			},
+			"ownerId": {
+				Type:        schema.TypeString,
+				Description: "The scorecard's owner.",
+				ForceNew:    false,
+				Required:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "The scorecard's description.",
+				ForceNew:    false,
+				Optional:    true,
+			},
+			"filterId": {
+				Type:        schema.TypeString,
+				Description: "The scorecard's filter.",
+				ForceNew:    false,
+				Optional:    true,
+			},
+
+			// computed fields
+			"aliases": {
+				Type:        schema.TypeString,
+				Description: "The scorecard's aliases.",
+				Computed:    true,
+			},
+			"filter": {
```

Comment:
@rocktavious `filter` and `owner` are full objects, so is `TypeString` appropriate here? Or do we need some kind of nested schema type?

Also, they are the same as `filterId`, `ownerId` except the full object. Should we still include them?

💭

Reply:
We don't need them we only need to read the filter and owner id into the `filterId` and `ownerId` fields in the read function.